webview-ui/src/panes/Charts.tsx

```diff
@@ -11,46 +12,106 @@ type ChartsProps = {
+          feature1={features.nominal[0]}
+          feature2={features.nominal[1]}
+        />
+      ) : (
+        [
+          ...features.nominal.map((x) => (
+            <Card key={`bucket-${x}`}>
+              <NominalChart
+                feature={x}
+                dataset={dataset}
+                viewValue={(value) => props.setFilteredView({ nominal: x, value })}
+              />
+            </Card>
+          )),
+          ...features.ordinal.flatMap((x) => [
+            <Card key={`feature-${x}`}>
+              <OrdinalChart
+                feature={x}
+                dataset={dataset}
+                viewValue={(value) => props.setFilteredView({ ordinal: x, value })}
+              />
+            </Card>,
+          ]),
+        ]
+      )}
+      {/* <Card className="text-sm">
       <strong>NOTE:</strong> You can add your own charts by registering "features" for your test
       inputs. Consult your PBT framework's documentation to learn more.
     </Card> */}
+      <Card>
+        <UniqueTimeChart dataset={dataset} />
+      </Card>
+    </div>
+  );
+};
+
+type NominalMosaicProps = {
+  samples: SampleInfo[];
+  setExampleFilter: (filter: ExampleFilter) => void;
+  feature1: string;
+  feature2: string;
+};
+
+const NominalMosaic = (props: NominalMosaicProps) => {
+  const { samples, setExampleFilter, feature1, feature2 } = props;
+
+  // Collect all unique values for each feature
+  const buckets1 = Array.from(
+    new Set(samples.map((x) => x.features.nominal[feature1]).filter((x) => x !== undefined))
+  );
+  const buckets2 = Array.from(
+    new Set(samples.map((x) => x.features.nominal[feature2]).filter((x) => x !== undefined))
+  );
+
+  // Presentation heuristic: the feature with fewer distinct categories should be the horizontal axis.
+  // It's not usually the case that there is more horizontal screen space than vertical.
+  const [[horizontalFeature, horizontalBuckets], [verticalFeature, verticalBuckets]] =
+    buckets1.length < buckets2.length
+      ? [
+          [feature1, buckets1],
+          [feature2, buckets2],
+        ]
+      : [
+          [feature2, buckets2],
+          [feature1, buckets1],
+        ];
+
+  return (
     <Card>
-      <UniqueTimeChart
-        dataset={dataset}
+      <span className="font-bold">Breakdown of </span> <span className="font-mono">{feature1}</span>{" "}
+      <span className="font-bold">vs.</span> <span className="font-mono">{feature2}</span>
+      <MosaicChart
+        samples={samples}
+        setExampleFilter={setExampleFilter}
+        horizontalAxis={horizontalBuckets.map((bucket) => [
+          bucket,
+          (x) => x.features.nominal[horizontalFeature] === bucket,
+        ])}
+        verticalAxis={verticalBuckets.map((bucket) => [
+          bucket,
+          (x) => x.features.nominal[verticalFeature] === bucket,
+        ])}
+        colors={["accent", "accent2", "accent3", "warning", "primary"]}
```

Comment:
I had trouble with colors. Ideally we'd have a bigger palette than 5. Currently if you have more than 5 rows, something bad will happen (no background color at all). We could add more colors, or I could make the colors repeat.

Reply:
I vote both, maybe? I'm using the [nord](https://www.nordtheme.com/docs/colors-and-palettes) color palette, so we can probably pull a few more colors from there, and then I think we should probably also have it repeat.